### PR TITLE
Correctly forward Host header in healthcheck

### PR DESCRIPTION
**Pull request by kamaradclimber**
Host header must be set explicitely on http requests

I'll write test a bit later

### CODE REVIEW

File: agent/check_test.go [hunk 1]
```diff
@@ -234,7 +235,9 @@ func TestCheckHTTP(t *testing.T) {
 					"User-Agent":      []string{"Consul Health Check"},
 				}
 				for k, v := range tt.header {
-					expectedHeader[k] = v
+                    if k != "Host" {
```
Comment (kamaradclimber) [inline agent/check_test.go]: thanks!

File: agent/check_test.go [hunk 2]
```diff
@@ -213,6 +213,7 @@ func TestCheckHTTP(t *testing.T) {
 
 		// custom header
 		{desc: "custom header", code: 200, header: http.Header{"A": []string{"b", "c"}}, status: api.HealthPassing},
+		{desc: "host header", code: 200, header: http.Header{"Host": []string{"customhost"}}, status: api.HealthPassing},
```
Comment (kamaradclimber) [inline agent/check_test.go]: done

**Comment by kamaradclimber** [comment]
thanks @magiconair for your comments (you've simply rewritten my patch :) )

File: agent/check_test.go [hunk 3]
```diff
@@ -234,8 +235,17 @@ func TestCheckHTTP(t *testing.T) {
 					"User-Agent":      []string{"Consul Health Check"},
 				}
 				for k, v := range tt.header {
-					expectedHeader[k] = v
+                    expectedHeader[k] = v
```
Comment (magiconair) [inline agent/check_test.go]: pls run `gofmt`. This doesn't look right.

Comment (kamaradclimber) [inline agent/check_test.go]: done thanks

**Comment by magiconair** [comment]
This LGTM. @slackpad do you want to have a quick peek?

**Comment by slackpad** [comment]
Looks good - thanks!

**Pull request merged**